File: parciales/ED1-Parcial1-2021-1--002/codigo/LCS.py
def mainSolve( s,  t):
    return solve(s, t, len(s), len(t)) 
  
def solve(s, t, i, j):
    if i == 0 or j == 0: 
       return 0;
    if s[i - 1] == t[j - 1]:
      return 1 + solve(s, t, i - 1, j - 1)
    else:
      return max(solve(s, t, i, j - 1), solve(s, t, i - 1, j))

def __main__():
    print (mainSolve("aa", "xayaz"))

__main__()

File: parciales/ED1-Parcial1-2021-1--002/codigo/LCS.java
class LCS{
  public static int solve(String s, String t){
    return solve(s, t, s.length(), t.length()); 
  }
  public static int solve(String s, String t, int i, int j){
    if(i == 0 || j == 0) 
      return 0;
    if(s.charAt(i - 1) == t.charAt(j - 1))
      return 1 + solve(s, t, i - 1, j - 1);
    //else
    return Math.max(solve(s, t, i, j - 1), solve(s, t, i - 1, j));
  }
  public static void main(String[] args){
    System.out.println(solve("aa", "xayaz"));
  }

}
